Automattic/woocommerce-payments
# Add compatibility for Multi-Currency and FedEx plugin.

Fixes #3128 

#### Changes proposed in this Pull Request

This PR is set to merge into the branch in https://github.com/Automattic/woocommerce-payments/pull/3133 due to it uses the same structure described in that PR, along with using some of the code added.

* New `WooCommerceFedEx` class in the `WCPay\MultiCurrency\Compatibility` namespace.
  * Uses the `wcpay_multi_currency_should_return_store_currency` to determine whether to return the store currency or the selected currency.
* Add or update tests as needed.

#### Testing instructions

* You will need to install the FedEx plugin and set it up.
  * It can be found in the `all-plugins` repo.
  * The creds for setup can be found in the secret store, and they go under WooCommerce > Settings > Shipping > FedEx.
  * If you haven't set up shipping before, you can add a new shipping zone, and if you don't specify a _Zone region_, it works for everywhere.
  * Once FedEx is added to your shipping zone, you can set the Origin Postcode to 34480. (It's what I use) https://d.pr/i/RnpOcr
  * Make sure all _Services_ are enabled. https://d.pr/i/XZuTKO
  * That should be it.
* You will need to have a product that has dimensions and a weight set on it. I typically do 5x5x5 for dimensions and a weight of 5.
* Add the item to your cart, and then go to the cart page.
* Check your shipping costs in your store currency against other currencies to make sure to prices are converted correctly.

-------------------


**Post merge**
